extensions/functions_arithmetic.yaml

```diff
@@ -181,6 +181,67 @@ scalar_functions:
           - value: i64
           - value: i64
         return: i64
+  -
+    name: "abs"
+    description: >
+      Calculate the absolute value of the argument.
+
+      Integer values allow the specification of overflow behavior to handle the
+      unevenness of the twos complement, e.g. Int8 range [-128 : 127].
+    impls:
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
+            required: false
+          - value: i8
+        return: i8
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
+            required: false
+          - value: i16
+        return: i16
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
+            required: false
+          - value: i32
+        return: i32
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
+            required: false
+          - value: i64
+        return: i64
+      - args:
+          - value: fp32
+        return: fp32
+      - args:
+          - value: fp64
+        return: fp64
+  -
+    name: "sign"
+    description: >
+      Return the signedness of the argument.
+
+      Integer values return signedness as Int8.
+      Floating point values return signedness with the same type as the input.
```

Comment:
I guess for ints the return value is one of [-1, 0, 1]. But what about floats? Would it be [-1.0, -0.0, 0.0, 1.0, NaN] or something?

Also, why do the integer variants return all `i8` but the floating point variants don't all return `fp32`?

Reply:
Yep.